leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/QueueModeTest.java

```diff
@@ -0,0 +1,213 @@
+/*******************************************************************************
+ * Copyright (c) 2017 RISE SICS AB.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     RISE SICS AB - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.leshan.integration.tests;
+
+import static org.junit.Assert.assertArrayEquals;
+
+import org.eclipse.leshan.Link;
+import org.eclipse.leshan.core.request.ReadRequest;
+import org.eclipse.leshan.core.response.ReadResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class QueueModeTest {
+
+    protected QueueModeIntegrationTestHelper queueModeHelper = new QueueModeIntegrationTestHelper();
+
+    @Before
+    public void start() {
+        queueModeHelper.initialize();
+        queueModeHelper.createServer((int) queueModeHelper.awaketime * 1000);
+        queueModeHelper.server.start();
+        queueModeHelper.createClient();
+    }
+
+    @After
+    public void stop() throws InterruptedException {
+        queueModeHelper.client.destroy(true);
+        queueModeHelper.server.destroy();
+        queueModeHelper.dispose();
+    }
+
+    @Test
+    public void awake_sleeping_awake_sleeping() {
+        // Check client is not registered
+        queueModeHelper.assertClientNotRegisterered();
+
+        // Start it and wait for registration
+        queueModeHelper.client.start();
+        queueModeHelper.waitForRegistration(1);
+
+        // Check client is well registered
+        queueModeHelper.assertClientRegisterered();
+        assertArrayEquals(Link.parse("</>;rt=\"oma.lwm2m\",</1/0>,</2>,</3/0>,</2000/0>".getBytes()),
+                queueModeHelper.getCurrentRegistration().getObjectLinks());
+
+        // Check that client is awake and only one notification
+        queueModeHelper.ensureClientAwake();
+
+        // Wait for client awake time expiration (1% margin)
+        queueModeHelper.waitForAwakeTime(queueModeHelper.awaketime * 1010);
+
+        // Check that client is sleeping
+        queueModeHelper.ensureClientSleeping();
+
+        // Wait for update when waking up (1% margin)
```

Comment:
I feel this could be clearer, if we trigger the update manually instead of the waitForUpdate.
This need to be aware and understand the schedule stuff in QueueModelHelper

Reply:
Ok, I'll change it and directly call `queueModeHelper.client.triggerRegistrationUpdate();` for waking up.